### PR TITLE
Replaces ActorIDs with Addresses where appropriate

### PR DESCRIPTION
**Pull request by alexytsu**

### CODE REVIEW

**Comment by alexytsu** [comment]
I'm unsure if `readonly` methods such as `balance_of` `allowance` should cause uninitialized accounts to be created

**Comment by anorth** [comment]
> I'm unsure if readonly methods such as balance_of allowance should cause uninitialized accounts to be created

No I think not.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:39]:
```suggestion
    /// Resolves the given address to its ID address form
```

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:67]: Nit: I think I'd like to rename `value` to `amount` everywhere, as in "amount of tokens". The VM will continue to use `value` for native tokens only.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:68]: For a follow-up PR, replace `&[u8]` with RawBytes everywhere it's used as receiver data, which will remove some noise here and elsewhere.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:40]: I think it's worth documenting the concrete "error" returned when the address is not resolvable, so that caller can catch that one explicitly while passing on others. Ditto below.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:100]: In the real implementation, ID, SECP, and BLS should all succeed always, and Actor always fail. I think we should replicate that here because callers may depend on that behaviour.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:107]: Probably not an issue right now, but really these should be variables so we can return distinct ids for different addresses. In fact they should be a map of already resolved ones so they are stable. This will probably bite someone at some point.  You can defer to #33 if you like.

File: fil_fungible_token/src/runtime/messaging.rs
Comment (anorth) [inline fil_fungible_token/src/runtime/messaging.rs:96]: I feel like there will make test-writing a little fiddly, and cause wide churn if we ever want to change these behaviours. All pushing towards either a much more robust fake, that can be configured, or a mock (#33).

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:135]: Even though we know it will be an ID address, take `Address` here because that's what the runtime gives us. Put the `.id().map_err()` noise inside the token library rather than require it at call sites.

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:202]: Use get_id_addr here – we don't want to create the actor. If it doesn't have an ID address it can't have a balance.

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:212]: `get_id_addr` both of these for similar reason. increase_allowance will have initialised if it was called, otherwise allowance must be zero.

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:224]: Address here and decrease_allowance too

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:269]: `get_id_addr`

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:119]: Nit: perhaps drop `addr` from the end of these names since they return IDs.
- resolve_addr_id
- get_addr_id
?

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:300]: Address

File: fil_fungible_token/src/token/mod.rs
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:356]: Address here too

**Review by anorth — COMMENTED** [review]
Thanks this is great, just needs a bit more!

**Comment by alexytsu** [comment]
@anorth this is ready for re-review
I'm deferring https://github.com/helix-onchain/filecoin/pull/46#discussion_r936132262
and https://github.com/helix-onchain/filecoin/pull/46#discussion_r936131896
to followups

**Review by anorth — APPROVED** [review]
Approving so we can roll forward, this is basically right.

There's still something I think we need to figure out about the intersections of uninitialized accounts and zero transfers. Let's discuss it in Slack.

**Pull request merged**
